### DISCUSSION TITLE
Deploy Keys - Expiration?

### DISCUSSION THREAD

**Discussion by justinodunn** [community/community#22513] [New to GitHub]
<p>Hi all! Just a super quick question. Do Github deploy keys expire? I know SSH keys are deleted automatically after 1 year of inactivity right?</p>
<p>Thank you!</p>

**Comment by queenofcorgis** [comment]
<p>Welcome to the forum, <a class="mention" href="/justinodunn">@justinodunn</a>! GitHub deploy keys do not expire and you are correct, <a href="https://docs.github.com/en/github/authenticating-to-github/deleted-or-missing-ssh-keys">SSH keys are deleted if they haven’t been used in a year</a>.</p>

**Comment by justinodunn** [comment]
<p>Thanks for the reply! Deploy Keys are SSH keys right? So Deploy Keys will also get deleted if they haven’t been used in a year?</p>

**Reply by ryankopf** [reply]
Yes.

**Comment by queenofcorgis** [comment, marked as answer]
<p><a class="mention" href="/justinodunn">@justinodunn</a> you’re welcome, glad I could help clarify. Also, confirming that deploy keys are SSH keys so they would expire after a year of being inactive.</p>

**Comment by mustafaStakater** [comment]
We are using deploy keys for our repositories which is then being used by ArgoCD, which simply clones the repository. Deploy keys were automatically removed from a few repositories suddenly.

Is there a clear description of what counts as activity and what doesnt?

**Comment by TomerFi** [comment]
Deploy Keys and Personal Access Tokens are disappearing, and since yesterday, I also have had issues validating webhook secrets.
https://github.com/orgs/community/discussions/107789

**Comment by antonosmond** [comment]
@queenofcorgis sorry to dig up an old thread but when you say:

> GitHub deploy keys do not expire

Is that accurate? Do they sometimes get removed rather than expire? It's my understanding that the validity of a deploy key depends on how that deploy key was added to the repo.

Please could you clarify the following for me and for others who come across this post:

1. A deploy key is added to a Github repo by a user with access to the repo. The user is then removed from the repo (account deleted, removed from members or removed from org or whatever). What happens to the deploy key?

2. A deploy key is added to a repo via the Github API, authenticated with a Github PAT. The PAT expires or is revoked. What happens to the deploy key? I'm pretty sure in this case the deploy is removed when the PAT expires so this may be why people have issues with disappearing deploy keys

3. A deploy key is added to a repo via the Github API, authenticated with a Github app installation token. How long is the deploy key valid for given the installation token is only valid for 1 hour? Are there any scenarios where this token will be removed e.g. uninstall of Github app or rotation of Github app private key?

Answers to these and any other scenarios you can think of would be super helpful! Documentation that describes these scenarios would be even better. If it already exists, please could you link to it?

Thank you so much!

**Reply by ArneRiemann4711** [reply]
I've just noticed the same thing, my PAT had expired and the DEPLOY keys set with it had subsequently disappeared. Is this a coincidence?
